### PR TITLE
Cray OpenMP wrapper flag

### PR DESCRIPTION
**Pull request by gppezzi**
First attempt to fix the "-True" flag that is added when using openmp as TC option on Cray. 
@boegel @pforai

### CODE REVIEW

**Comment by boegel** [comment]
@gppezzi looks perfect to me... Did you test it? Any example easyconfigs where you have `openmp` enabled?

**Comment by gppezzi** [comment]
I've tested with GROMACS GPU but this is WIP, I will try with other simpler cases and let you know.

**Comment by boegel** [comment]
@gppezzi maybe try building `STREAM` with the Cray* toolchains to test this?

**Comment by pforai** [comment]
@gppezzi, @boegel 

The OpenMP flags are sadly inconsistent between the drivers for CCE, Intel, GNU and PGI:

```bash
OpenMP support:
   -homp		Enables OpenMP and links in OpenMP libraries when
			possible using CCE.
			(This is the default.)

   -hnoomp		Disables OpenMP and links in non-OpenMP libraries
			when using CCE.

   -fopenmp		Enables OpenMP and links in OpenMP libraries when
			possible using GNU.

   -openmp		Enables OpenMP and links in OpenMP libraries when
			possible when using Intel.

   -mp			Enables OpenMP and links in OpenMP libraries when
			possible using PGI.

   -Mnoopenmp		Disables OpenMP and links in non-OpenMP libraries
			when using PGI.
```

**Comment by gppezzi** [comment]
If the new implementation makes sense to you, I'll apply the same fix accordingly to Intel, CCE and PGI. Let me know.

``` 
eb ~/GROMACS-5.1.2-CrayGNU-2016.03-cuda-7.0.eb -x | grep CFLAGS
```
* Before the fix
```
  export CFLAGS="-craype-verbose -True -O2"
  export FCFLAGS="-craype-verbose -True -O2"
```
* After
```
  export CFLAGS="-craype-verbose -fopenmp -O2"
  export FCFLAGS="-craype-verbose -fopenmp -O2"
```

**Comment by pforai** [comment]
Yeah, this looks good! 

Update the unique compiler map for each of the subclasses accordingly.

**Comment by gppezzi** [comment]
I've updated the other subclasses. 

So far I've tested this for GNU and Intel. CCE and PGI still need to be checked.

File: easybuild/toolchains/crayintel.py [hunk 1]
```diff
@@ -38,3 +38,4 @@ class CrayIntel(CrayPEIntel, CrayMPICH, LibSci):
     """Compiler toolchain for Cray Programming Environment for Intel compilers (PrgEnv-intel)."""
     NAME = 'CrayIntel'
     SUBTOOLCHAIN = DUMMY_TOOLCHAIN_NAME
+    CrayPEIntel.COMPILER_UNIQUE_OPTION_MAP.update({'openmp': 'openmp'})
```
Comment (boegel) [inline easybuild/toolchains/crayintel.py]: `openmp` is deprecated for recent Intel versions, better stick to `fopenmp` here too, cfr. #1718

**Comment by boegel** [comment]
@gppezzi I think you'll need to tackle this in `easybuild/toolchains/compiler/craype.py` instead, where we already fiddle with `self.COMPILER_UNIQUE_OPTION_MAP` for the precision flags (which are also compiler specific)

**Comment by gppezzi** [comment]
@boegel now tested with CCE, Intel and GNU (I don't have any test for PGI)
cc @pforai @jgphpc 

 * CCE
```
export CFLAGS="-craype-verbose -homp -O2"
```
* Intel
```
export CFLAGS="-craype-verbose -fopenmp -O2 -ftz -fp-speculation=safe -fp-model"
```
* GNU
```
export CFLAGS="-craype-verbose -fopenmp -O2"
```

File: easybuild/toolchains/compiler/craype.py [hunk 2]
```diff
@@ -137,6 +137,7 @@ class CrayPEGCC(CrayPECompiler):
     def __init__(self, *args, **kwargs):
         """CrayPEGCC constructor."""
         super(CrayPEGCC, self).__init__(*args, **kwargs)
+        self.COMPILER_UNIQUE_OPTION_MAP.update({'openmp': 'fopenmp'})
```
Comment (boegel) [inline easybuild/toolchains/compiler/craype.py]: @gppezzi I would go with this instead, just to match the style below:

```python
self.COMPILER_UNIQUE_OPTION_MAP['openmp'] = 'fopenmp'
```

other than that: should be good to go

**Comment by gppezzi** [comment]
@boegel  any other remark?

**Comment by boegel** [comment]
lgtm, thanks @gppezzi

**Pull request merged**